### PR TITLE
[optimize] Add CUDA Graph Support for S2 Pro Model

### PR DESCRIPTION
**Pull request by JingwenGu0829**
Unify the S2-Pro slow head (text model) and fast head (codebook decoder) into a single `model.forward()` call, enabling end-to-end CUDA graph capture for the entire decode step.

### Key changes

**Unified decode pipeline** — Previously, the text model ran batched forward via SGLang, then `S2ProSGLangOutputProcessor` ran the codebook loop per-request in a separate stage. Now both stages are fused: `S2ProSGLangTextModel.forward()` runs transformer → constrained semantic sampling → batched codebook generation (`_decode_codebooks`) all in one call.

**CUDA graph covers the full step** — The factory now defers CUDA graph capture (`disable_cuda_graph=True` during `ModelWorker.__init__`) until `setup_vq_decode()` attaches the audio decoder and allocates persistent GPU buffers. Then `init_device_graphs()` captures the complete decode path including `_decode_codebooks`.

**Persistent GPU buffers replace per-request processing** — Pre-allocated tensors (`_vq_codes`, `_vq_mask`, `_output_codes`, `_output_semantic_ids`) on the model serve as CUDA-graph-safe input/output interfaces. The `ModelRunner` writes VQ codes from the previous step into these buffers before forward, and reads output codes after forward — no per-request tensor construction needed.

**Batched codebook generation** — The codebook loop now processes all requests in the batch simultaneously (matrix ops over `[bs, ...]` tensors) instead of one request at a time.

**VQ embedding combination inlined into the model** — During decode, instead of calling `audio_decoder.embed_one_token()` externally, the model combines text embeddings with codebook embeddings directly using the shared `codebook_embeddings` table, keeping everything within the CUDA graph boundary.

### Removed

- `S2ProSGLangOutputProcessor` class (its logic now lives in `_decode_codebooks` and persistent buffers)
- `_codebook_loop_impl` standalone function
- `_sample_with_topk` import (semantic sampling now uses `torch.argmax` on biased logits)
- `use_torch_compile` parameter from factory

### Files changed

| File | What changed |
|------|-------------|
| `sglang_model.py` | Added `setup_vq_decode()`, `_decode_codebooks()`, VQ combination in `forward()`, persistent buffers |
| `s2pro_sglang_ar.py` | Removed `S2ProSGLangOutputProcessor`, split `_inject_vq_embeds` → `_inject_vq_embeds_prefill` + `_update_vq_buffers`, added `_build_outputs` reading from model buffers |
| `factory.py` | Deferred CUDA graph capture, removed `use_torch_compile` param, calls `setup_vq_decode` + `setup_caches` before graph capture |
| `stages.py` | Enabled CUDA graph by default (`disable_cuda_graph=False`), simplified codec loading |

### Why `torch.compile` is not included

During the development of this optimization, we explored `torch.compile` as an additional acceleration path. Benchmarking by [@Ratish1](https://github.com/Ratish1) in [PR #153](https://github.com/sgl-project/sglang-omni/pull/153) showed the following results:

| Configuration | Startup time | Steady-state throughput |
|---|---|---|
| No compile (CUDA graph only) | ~33s | 88 tok/s |
| Partial compile (fast head only) | ~54s | 121 tok/s |
| Full-model compile | ~137s | 126 tok/s |

Full-model compile adds ~100s startup overhead for only ~4% throughput gain over partial compile — not a good default. Partial compile targeting only the codebook path offers a better tradeoff, but introducing `torch.compile` at the model level brings significant complexity:

1. **Maintainability** — `torch.compile` requires careful graph-break management. Any future change to `_decode_codebooks` or the VQ combination logic must be verified for compile compatibility (`fullgraph=True`), creating an invisible maintenance burden.
2. **Debuggability** — Compiled code produces opaque stack traces. For a pipeline with as many moving parts as S2-Pro (paged KV cache + static KV cache + codebook loop + VQ embeddings), keeping the eager path readable is more valuable at this stage.
3. **Framework-level concern** — `torch.compile` optimization belongs at the SGLang framework level (e.g., mega cache), not hard-wired into model-specific code. When SGLang adds first-class `torch.compile` support, S2-Pro will benefit automatically without model-side complexity.

For these reasons, this PR focuses on the architectural win (unified CUDA graph) and defers `torch.compile` to future framework-level work.

### CODE REVIEW

File: benchmarks/benchmark_s2pro_speed.py [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+    codebook_size: int,
+    max_new_tokens: int = 2048,
+    top_k: int = 30,
+):
+    from sglang.srt.server_args import ServerArgs
+
+    from sglang_omni.models.fishaudio_s2_pro.factory import (
+        _patch_fish_config_for_sglang,
+        create_s2pro_sglang_engine,
+    )
```
Comment (zhaochenyang20) [inline benchmarks/benchmark_s2pro_speed.py]: move all the import to the beginning of the file.
